src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -344,7 +343,19 @@ private void writeValues(List<Object> values, GroupType type) {
                     "Unsupported column type: " + cols.get(i).asPrimitiveType());
             }
           } else {
-            throw new ParquetEncodingException("Unsupported column type: " + cols.get(i));
+            if (cols.get(i).isRepetition(Type.Repetition.REPEATED)) {
+              for (Object o : values) {
+                recordConsumer.startGroup();
+                if (o != null) {
+                  writeValues((List<Object>) o, cols.get(i).asGroupType());
```

Comment:
how are we sure that this cast will not fail here?
also applies to the cast in line 356.

Reply:
Cast to `List<?>`